src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -111,13 +184,48 @@ public async ValueTask DisposeAsync()
 
             await OnTick().ConfigureAwait(false);
 
+            if (ReferenceEquals(_batchedLogEventSink, this))
+            {
+                // The sink is being used in the obsolete inheritance-based mode. Old sinks won't
+                // override something like `DisposeAsyncCore()`; we just forward to the synchronous
+                // `Dispose()` method to ensure whatever cleanup they do still occurs.
+                Dispose(true);
+                return;
```

Comment:
This was the original stumbling block that prevented me from enabling `IAsyncDispose` along with the legacy inheritance-based API.